drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/editor/GuidedRuleEditorPresenter.java

```diff
@@ -93,6 +96,9 @@
     @Inject
     private ValidationPopup validationPopup;
 
+    @Inject
+    private Instance<RuleModellerActionPlugin> actionPluginInstance;
```

Comment:
You **might** be better off using ```ManagedInstance<..>``` to avoid you otherwise having to destroy instances you create. The ```GuidedRuleEditorPresenter``` is ```@Dependent``` scoped bean and hence a new instance will be created every time the editor is opened. Unless you **guarantee** ```RuleModellerActionPlugin``` instances are always ```@ApplicationScoped``` you could have a leak.